File: app/src/main/java/com/widiarifki/screentest/presentation/home/HomeFragment.java
package com.widiarifki.screentest.presentation.home;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import com.widiarifki.screentest.MainActivity;
import com.widiarifki.screentest.R;
import com.widiarifki.screentest.presentation.menu.MenuFragment;

/**
 * Created by widiarifki on 27/02/2018.
 */

public class HomeFragment extends Fragment implements IHomeView {

    private HomePresenter homePresenter;
    private Context mContext;
    private MainActivity mMainActivity;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = getContext();
        mMainActivity = (MainActivity)mContext;
        mMainActivity.showActionBar(false);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home, container, false);
        homePresenter = new HomePresenter(this);

        final EditText inputName = (EditText) view.findViewById(R.id.inputName);
        Button btnEnter = (Button) view.findViewById(R.id.btnEnter);
        btnEnter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = inputName.getText().toString();
                mMainActivity.setName(name);
                homePresenter.checkPalindrome(name);
                goToMenu();
            }
        });

        return view;
    }

    @Override
    public void showDialogue(String title, String info) {
        new AlertDialog.Builder(mContext)
                .setTitle(title)
                .setMessage(info)
                .create()
                .show();
    }

    @Override
    public void goToMenu() {
        mMainActivity.setFragment(new MenuFragment(), MenuFragment.TITLE);
    }
}

File: app/src/main/java/com/widiarifki/screentest/presentation/home/HomeFragment.kt
package com.widiarifki.screentest.presentation.home

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText

import com.widiarifki.screentest.MainActivity
import com.widiarifki.screentest.R
import com.widiarifki.screentest.presentation.menu.MenuFragment

/**
 * Created by widiarifki on 27/02/2018.
 */

class HomeFragment : Fragment(), IHomeView {

    private var homePresenter: HomePresenter? = null
    private var mContext: Context? = null
    private var mMainActivity: MainActivity? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = context
        mMainActivity = mContext as MainActivity?
        mMainActivity!!.showActionBar(false)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_home, container, false)
        homePresenter = HomePresenter(this)

        val inputName = view.findViewById(R.id.inputName) as EditText
        val btnEnter = view.findViewById(R.id.btnEnter) as Button
        btnEnter.setOnClickListener {
            val name = inputName.text.toString()
            mMainActivity!!.name = name
            homePresenter!!.checkPalindrome(name)
            goToMenu()
        }

        return view
    }

    override fun showDialogue(title: String, info: String) {
        AlertDialog.Builder(mContext!!)
                .setTitle(title)
                .setMessage(info)
                .create()
                .show()
    }

    override fun goToMenu() {
        mMainActivity!!.setFragment(MenuFragment(), MenuFragment.TITLE)
    }
}
